kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1398 @@
+                              pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                              x-kubernetes-int-or-string: true
+                            maxNetworkCost:
+                              type: integer
+                              default: 1
+                              minimum: 0
+                              maximum: 100
+                              format: int64
+                              description: The maximum network cost between both Pods.
+                            required:
+                              - podName
+          status:
+            description: Record Pod Allocations (Pod id, hostnames).
+            properties:
+              runningPods:
+                description: The number of actively running pods.
+                format: int32
+                type: integer
+                minimum: 0
+              scheduled:
+                type: array
+                additionalProperties:
+                  podName:
+                    type: string
+                    description: Name of the Pod.
+                  replicaID:
+                    type: string
+                    description: Pod ID
+                  hostname:
+                    type: string
+                    description: Hostname allocating Pod instance
+              scheduleStartTime:
+                description: ScheduleStartTime of the AppGroup
+                format: date-time
+                type: string
+              topologyCalculationTime:
+                description: topologyCalculationTime of the AppGroup
+                format: date-time
+                type: string
+              topologyOrder:
+                description: The optimal order to schedule pods on this App Group based on a given algorithm.
+                type: array
+                items:
+                  description: Pod Name and Pod Priority index
+                  type: object
+                    podName:
+                      type: string
+                      description: Name of the Pod.
+                    index:
+                      type: integer
+                      default: 1
+                      minimum: 1
+                      format: int64
+                      description: Priority index for each Pod (e.g., 1, 2, ...)
+                        (1 means pod should be scheduled first in the AppGroup)
+            type: object
+        type: object
+    served: true
+    storage: true
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+```
+
+### Example
+
+<p align="center"><img src="figs/chain.png" title="Chain" width="600" class="center"/></p>
+
+```yaml
+# Example App Group CRD spec
+apiVersion: scheduling.sigs.k8s.io/v1alpha1
+kind: AppGroup
+metadata:
+  name: A1
+spec:
+  numMembers: 3
+  topologySortingAlgorithm: KahnSort
+  Pods:
+    - podName: P1
```

Comment:
The `podname` refers to the deployment metadata name and not to the pod replica id. This part allows us to establish a service dependency graph. If the pod name changes, the AppGroup CRD must be updated by the user/operator. By letting users specify their service topology graph, we can calculate the preferred order to allocate the pods forming an AppGroup. 

We will modify the `podname` field to `name` and also change `Pods` to `Resources` since we also plan to target ReplicaSets, StatefulSets, etc. 

Without the service topology graph, all pods will be considered by the plugins in terms of service dependency, which might not be true since some pods might not interact with each other...